### PR TITLE
Added information on custom docker configurations

### PR DESCRIPTION
**Pull request by drpayyne**
## Purpose of this pull request

This pull request (PR) adds information on how custom docker configurations work. This is in regard to the issue: https://github.com/magento/magento-cloud-docker/issues/94

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-quick-reference.html
-  https://devdocs.magento.com/guides/v2.2/cloud/docker/docker-quick-reference.html
-  https://devdocs.magento.com/guides/v2.3/cloud/docker/docker-config.html
-  https://devdocs.magento.com/guides/v2.2/cloud/docker/docker-config.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by drpayyne** [comment]
@hguthrie I changed the base branch as you suggested and the changed file count shot up... How do I proceed?

**Comment by hguthrie** [comment]
I suggested using the Cloud-Docker-1.0.0 release branch instead of master because we expect to make significant changes to the Docker documentation in this next release. If you choose to change the base branch instead of creating a new branch from the Cloud-Docker-1.0.0 release branch, then I would always recommend a compare first.

In this case, I went ahead and updated the Cloud-Docker-1.0.0 release branch with master and ran a compare to your working branch:

https://github.com/magento/devdocs/compare/magento:Cloud-Docker-1.0.0...drpayyne:docker-issue-94

The compare appears normal to me, showing the expected file conflicts that you would need to address.

I am not sure why this PR is showing so many commits. My only conclusion would be that there is some branch in the chain that is out of date and so the commit history is confused.

**Comment by drpayyne** [comment]
Resolving the conflicts, fixed the incorrect commit history as the merge wasn't completed until I resolved the conflicts.

File: guides/v2.2/cloud/docker/docker-config.md [hunk 1]
```diff
@@ -9,6 +9,48 @@ functional_areas:
...
+| Redis         | `--redis`  | 3.2, 4.0, 5.0
+
+The `docker:build` command runs in interactive mode and verifies the configured service versions. To skip the interactive mode, use the `-n, --no-interaction` option.
+
+For example, the following command starts the Docker configuration generator for the developer mode and specifies the PHP version 7.2:
+
+```bash
+./vendor/bin/ece-tools docker:build --mode="developer" --php 7.2
+```
+
```
Comment (drpayyne) [inline guides/v2.2/cloud/docker/docker-config.md]: Although @hguthrie, I did not submit these lines. These are already present and I only submitted a callout information _on top of_ these changes. Please check into this. Thanks!

Comment (hguthrie) [reply guides/v2.2/cloud/docker/docker-config.md]: No worries. I'll handle it. Thanks!

Comment (hguthrie) [reply guides/v2.2/cloud/docker/docker-config.md]: All righty, @drpayyne . It was just some dust left over from the merge conflict. The config topic changed quite a bit (and still may change for the release), but I cleaned it.

I edited your Warning. It is a helpful warning and I am glad you posted it. Let me know if your original intention was preserved.

I still need to continue the review, but I usually try out these things in my Cloud first, so I'll get back to it soon.

**Comment by shiftedreality** [comment]
As part of https://github.com/magento/magento-cloud-docker/issues/94 we should also adjust https://github.com/magento/magento-cloud-docker/blob/develop/dist/bin/magento-docker to automatically pick up these files.

But DevDocs PR is looking good.

**Pull request merged**